pkg/dev/podmandev/reconcile.go

```diff
@@ -118,6 +123,20 @@ func (o *DevClient) reconcile(
 	return nil
 }
 
+// warnAboutK8sResources prints a warning if the Devfile contains a K8s resource that it needs to create on Podman.
+func (o *DevClient) warnAboutK8sResources(devfileObj parser.DevfileObj) {
+	k8sComponents, err := devfile.GetKubernetesComponentsToPush(devfileObj, false)
```

Comment:
One more thing I just thought about: if the run command is a composite referencing an `apply` command with a K8s component, I think no warning will be displayed for this component.
I think you also need to consider this case by also displaying a warning in the command handler.